### DISCUSSION TITLE
Strange behaviour with Docker

### DISCUSSION THREAD

**Discussion by JABull1066** [Chaste/Chaste#187] [Q&A]
There seems to be some weird behaviour running cmake in the docker environment - build files always get written to /home/chaste/, rather than the directory from which cmake was called. Normally it'd generate files in the build directory where cmake is called, rather than in the chaste directory, as I* would expect.

Any ideas @bdevans ?

*By "I", I mean @fcooper8472 , who told me what to write here...

**Comment by bdevans** [comment]
Hi, what command are you using to call `cmake`?

**Comment by fcooper8472** [comment]
Hi @bdevans, I'll answer on @JABull1066 behalf as I was the one actually typing on his keyboard when we ran into the problem.

I created a new directory inside the main chaste directory, and ran:

```
cmake ..
```

as normal.  But that appeared to generate the Makefile in the main Chaste directory and not in the build directory. This is definitely bizarre. Any idea if this is related to the docker environment somehow?

**Reply by bdevans** [reply]
Well I always use the included scripts which call `cmake` with:
`cmake ... -H$CHASTE_SOURCE_DIR -B$CHASTE_BUILD_DIR`
and then `make` with:
`make --no-print-directory -j$NCORES -C $CHASTE_BUILD_DIR`.
The referenced variables are indeed set as Docker environment variables as follows:
```bash
ENV CHASTE_SOURCE_DIR="${CHASTE_DIR}/src" \
    CHASTE_BUILD_DIR="${CHASTE_DIR}/build" \
    CHASTE_PROJECTS_DIR="${CHASTE_DIR}/src/projects" \
    CHASTE_TEST_OUTPUT="${CHASTE_DIR}/testoutput"
```
Do the environment variables explain it?
